spec/features/adding_items_to_cart_spec.rb

```diff
@@ -155,6 +153,117 @@
+        bundled_product_from_options(
+          name: "Hat", option_type: "Color", option_values: ["Red", "Blue"]
+        )
+      end
+
+      before do
+        add_part_to_bundle(
+          bundle,
+          keychain.master,
+          count: 1
+        )
+
+        add_part_to_bundle(
+          bundle,
+          shirt.master,
+          variant_selection_deferred: true
+        )
+
+        add_part_to_bundle(
+          bundle,
+          hat.master,
+          variant_selection_deferred: true
+        )
+      end
+
+      context "and the user selects differing variants from the existing line item" do
+        specify "the cart contains incremented variants when listing bundle items" do
+          add_item_to_cart(size: "Large", color: "Red")
+          add_item_to_cart(size: "XL", color: "Blue")
+
+          within all("#cart-detail .line-item")[0] do
+            expect(page).to have_content(bundle.name)
+            expect(page).to have_css("input[value='1']")
+            expect(page).to(
+              have_content("(1) Keychain (KEYCHAIN)")
+            )
+            expect(page).to(
+              have_content("(1) Shirt (Size: Large) (SHIRT-LARGE)")
+            )
+            expect(page).to(
+              have_content("(1) Hat (Color: Red) (HAT-RED)")
+            )
+          end
+
+          within all("#cart-detail .line-item")[1] do
+            expect(page).to have_content(bundle.name)
+            expect(page).to have_css("input[value='1']")
+            expect(page).to(
+              have_content("(1) Keychain (KEYCHAIN)")
+            )
+            expect(page).to(
+              have_content("(1) Shirt (Size: XL) (SHIRT-XL)")
+            )
+            expect(page).to(
+              have_content("(1) Hat (Color: Blue) (HAT-BLUE)")
+            )
+          end
+        end
+      end
+
+      context "and the user selects the same variants as the existing line item" do
+        specify "the cart contains incremented variants when listing bundle items" do
+          2.times { add_item_to_cart(size: "Large", color: "Red") }
+
+          within "#cart-detail .line-item" do
+            expect(page).to have_content(bundle.name)
+            expect(page).to have_css("input[value='2']")
+            expect(page).to(
+              have_content("(2) Keychain (KEYCHAIN)")
+            )
+            expect(page).to(
+              have_content("(2) Shirt (Size: Large) (SHIRT-LARGE)")
+            )
+            expect(page).to(
+              have_content("(2) Hat (Color: Red) (HAT-RED)")
+            )
+          end
+        end
+      end
+    end
+  end
+
+  def add_item_to_cart(args)
+    visit spree.product_path(bundle)
+
+    select "Size: #{args[:size]}", from: "options_selected_variants_2"
+    select "Color: #{args[:color]}", from: "options_selected_variants_3"
+    click_button "add-to-cart-button"
+  end
+
+  def bundled_product_from_options(args)
+    args[:sku] ||= args[:name].parameterize.upcase
```

Comment:
It's just to provide a default, if you provide a sku nothing gets set, if you don't name becomes sku